Chainlit/literalai-python
# fix: the python sdk lacks docstring documentation

### Changes:
- made all doc part of BaseLiteralAPI so async versions get doc (previous hack with __doc__ didn't work at all)
- added typing to all APIs
- reworded all the documentation
- fixed misalignments between sync and async clients

### To test:
Open your favorite IDE and check you can view documentation for both the sync and async clients APIs / classes
All objects (prompt, dataset, etc. should have documentation). All methods in API should have documentation. 

Some formatting issues with Cursor, but it should render well on VSCode at least.
